test/tests/restart/restart_subapp_not_master/two_step_solve_sub_restart.i

```diff
@@ -0,0 +1,82 @@
+[Mesh]
+  file = two_step_solve_master_cp/0002_mesh.cpr
+[]
+
+[Problem]
+  restart_file_base = two_step_solve_master_cp/LATEST
+  allow_sub_app_restart = true
```

Comment:
I'm thinking of changing this flag to something else. It turns out it's not accurate for one because this flag appears in the sub_app file not the "ultimate master" file. I'm think the flag should say `force_restore` or `allow_local_restore` or maybe even `skip_use_master_backup` :smile:

I think I'll add an additional error message too like I mentioned earlier unless you are against this PR.